File: 5-kyu/valid-parentheses/ruby/solution.rb
def valid_parentheses(string)
  par1 = 0
  
  string.split("").each_with_index{ |c, index|
    puts(c)
    if (c == "(")
      par1+=1
    end
    if (c == ")" && par1 != 0)
      par1-=1
    elsif (par1 == 0 && c == ")")
      return false
    end
  }
  return (par1 == 0)
end

File: 5-kyu/valid-parentheses/javascript/solution.js
function validParentheses(parens) {
  let par1 = 0;
  for (let index in [...parens])
  {
      const c = [...parens][index]
      if (c == "(")
      {
        par1++;
      }
      if (c == ")" && par1 != 0)
      {
        par1--;
      }
      else if (par1 == 0 && c == ")")
      {
        return false
      }
  }
  return par1 == 0
}